docs/guide.ipynb

```diff
@@ -0,0 +1,180 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "mechanical-pointer",
+   "metadata": {},
+   "source": [
+    "# How to customize or contribute to the knowledge base\n",
+    "\n",
+    "The [asp](https://github.com/cmudig/draco2/tree/main/draco/asp) directory contains the Draco knowledge base as answer set programs. \n",
+    "\n",
+    "## draco asp files\n",
+    " * `define.lp` declares the domains to visualization attributes.\n",
+    " * `helper.lp` defines useful helper functions. \n",
+    " * `generate.lp` sets up the search space.\n",
+    " * `constraints.lp` restricts the search space to follow the correct draco general description language. \n",
+    " * `hard.lp` restricts the search space to only well-formed and expressive specifications.\n",
+    " * `soft.lp` defines soft constraints in the form of violation/1 and violation/2 predicates. By themselves, these predicates don't change the search.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "promising-steal",
+   "metadata": {},
+   "source": [
+    "## guideline\n",
+    "You can design your own description language and use it with Draco or extend the existing language we use here. If you don't know where to start with the constraints, you can first set up the search space in `generate.lp` and use our `run_clingo` API to generate some recommendations. Then, you should be able to find some recommendations that should have been left out, and you can write constraints to reflect them.  \n",
```

Comment:
Slight modify this so it doesn't assume that you can modify generate.lp.

Reply:
I didn't mean they can modify the generate.lp from the programs.
I'm thinking about when they design their own language, adding say entity types we don't define and writing their new generator. Is this outside of what you imagined for draco2?

Reply:
Your right that that's in scope. We already have a place to document existing programs, though, so I want to make sure we don't repeat content.